### PR TITLE
Milestone2: Bookstore add reducers and actions

### PR DESCRIPTION
**Pull request by Chandan-devs-tech**
## In this project I have done:
- Added Redux Toolkit (npm install react-redux @reduxjs/toolkit).
- Created a directory that contains all Redux logic (/src/redux)
- Configured a Redux store (/src/redux/store.js)
- Defined a slice of state for books that:
  - Included an array of books (initial state: empty array)
  - Included a reducer that adds a book
  - Includes a reducer that removes a book
- Defined a slice of state for categories that:
  - Includes an array of categories (initial state: empty array)
  - Includes a reducer that checks the status and always returns "Under construction".
- Structured my application files using a "feature folder" approach and used the [ducks pattern]
- NOTE: No changes in React components are required in this project.

### CODE REVIEW

File: src/redux/store.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
+
+export default store;
```
Comment (danmainah) [inline src/redux/store.js]: - [ ] @Chandan-devs-tech Great work adding your store however you forgot to import the books and categories reducers in your store.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const initialState = {
+  categories: [],
+};
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: () => 'Under construction',
+  },
```
Comment (danmainah) [inline src/redux/categories/categoriesSlice.js]: - [ ] @Chandan-devs-tech Great work on your categories reducer, however, the reducer needs to check the status if it matches the string "Under construction" and then return "Under construction" string.
```suggestion
  reducers: {
   checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Chandan-devs-tech , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Setup the redux directory using the [ducks pattern](https://github.com/erikras/ducks-modular-redux)
✔️  Added the reducers using createSlice.
✔️  Added the redux store


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Chandan-devs-tech👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**